velox/vector/DecodedVector.h

```diff
@@ -283,10 +265,20 @@ class DecodedVector {
   // complex type.
   const void* data_ = nullptr;
 
-  // Null bitmask. One bit for each T in 'data_'. nullptr f if there
-  // are no nulls.
+  // Null bitmask of the base vector if wrappings didn't add nulls
+  // (hasExtraNulls_ is false). Otherwise, null bitmask of the base vector
+  // combined with null bitmasks in all the wrappings (hasExtraNulls_ is true).
+  // May be nullptr if there are no nulls.
   const uint64_t* nulls_ = nullptr;
 
+  // Nulls bitmask containing null bits of the base vector combined with null
+  // bits in all the wrappings. May be nullptr if there are no nulls or
+  // allNullsInitialized_ is false. Initialized on first access.
+  const uint64_t* allNulls_ = nullptr;
```

Comment:
```
nit: how about mergedNulls_? Not sure.
```

Reply:
mergedNulls_ works. combinedNulls_ is an alternative. Let me know which one you prefer.

In a follow-up PR, I'd like to rename nulls_ to baseNulls_ and change the code to only store nulls from the base vector in baseNulls_ and only do that when wrappings do not add nulls. When wrappings add nulls, we would populate allNulls_ (or mergedNulls_) and set xxxInitialized_ flag to true. This might be a bit clearer, but refactoring itself will be messy, hence, I prefer to do it separately from the API change.